File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/calendar/AuditCalendarEventModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.calendar;

import java.util.List;

import ro.cloudSoft.cloudDoc.presentation.client.shared.model.OrganizationEntityModel;

public class AuditCalendarEventModel extends IntervalCalendarEventModel {
	
	public AuditCalendarEventModel() {
		super(CalendarEventType.AUDIT);
	}
	
	private String location;
	private List<OrganizationEntityModel> attendees;
	private String documentId;
	private String documentLocationRealName;
	
	public String getLocation() {
		return location;
	}
	public void setLocation(String location) {
		this.location = location;
	}
	public List<OrganizationEntityModel> getAttendees() {
		return attendees;
	}
	public void setAttendees(List<OrganizationEntityModel> attendees) {
		this.attendees = attendees;
	}
	public String getDocumentId() {
		return documentId;
	}
	public void setDocumentId(String documentId) {
		this.documentId = documentId;
	}
	public String getDocumentLocationRealName() {
		return documentLocationRealName;
	}
	public void setDocumentLocationRealName(String documentLocationRealName) {
		this.documentLocationRealName = documentLocationRealName;
	}
	
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/calendar/audit-calendar-event.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { OrganizationEntityModel } from "../organization-entity.model";
import { IntervalCalendarEvent } from "./interval-calendar-event.model";
import { StringUtils } from "@app/shared/utils";

@JsonObject
export class AuditCalendarEventModel extends IntervalCalendarEvent {

	@JsonProperty("location", String)
	public location: string = null;

	@JsonProperty("attendees", [OrganizationEntityModel])
	public attendees: OrganizationEntityModel[] = [];
	
	@JsonProperty("documentId", String)
	public documentId: string = null;
	
	@JsonProperty("documentLocationRealName", String)
	public documentLocationRealName: string = null;

	public get hasDocument(): boolean {
		return (StringUtils.isNotBlank(this.documentId) && StringUtils.isNotBlank(this.documentLocationRealName));
	}
}